### PR TITLE
Kubernetes API to take port from Lookup

### PR DESCRIPTION
**Pull request by chbatey**
Addresses a few gotchas when @alanngai was trying bootstrap

* Lookup contains a port, use that and fallback to the one in config
* Pick up namespace from environment variable if present
* Add some more logging to k8s api

### CODE REVIEW

File: bootstrap-demo/kubernetes-api/kubernetes/akka-cluster.yml [hunk 1]
```diff
@@ -28,13 +28,18 @@ spec:
...
-          protocol: TCP
-        # external http
-        - name: akka-mgmt-http
+        # must match up with contact-point-discovery.port-name for boostrap
+        - name: management
           containerPort: 8558
           protocol: TCP
+#namespace
+        env:
+        - name: NAMESPACE
```
Comment (chbatey) [inline bootstrap-demo/kubernetes-api/kubernetes/akka-cluster.yml]: This isn't mandatory, the config will default to "default" if this isn't set

File: bootstrap-demo/kubernetes-api/kubernetes/akka-cluster.yml [hunk 2]
```diff
@@ -28,13 +28,18 @@ spec:
           containerPort: 2552
           protocol: TCP
         # akka-management bootstrap
-        - name: bootstrap
-          containerPort: 8558
-          protocol: TCP
-        # external http
-        - name: akka-mgmt-http
+        # must match up with contact-point-discovery.port-name for boostrap
```
[the remark this answers is not in the document]
Comment (raboof) [reply bootstrap-demo/kubernetes-api/kubernetes/akka-cluster.yml]: ('boostrap' typo)

File: bootstrap-demo/kubernetes-api/src/main/resources/application.conf [hunk 3]
```diff
@@ -1,3 +1,6 @@
+akka.loglevel = "INFO"
```
Comment (raboof) [inline bootstrap-demo/kubernetes-api/src/main/resources/application.conf]: To set a good example, maybe use loglevel DEBUG and filter in slf4j/logback? OK though.

File: bootstrap-demo/kubernetes-api/src/main/scala/akka/cluster/bootstrap/DemoApp.scala [hunk 4]
```diff
@@ -14,10 +14,7 @@ import com.typesafe.config.ConfigFactory
 
 object DemoApp extends App {
 
-  implicit val system = ActorSystem("Appka", ConfigFactory.parseString("""
-       akka.loglevel = INFO
-       akka.actor.provider = cluster
-    """).withFallback(ConfigFactory.load()))
+  implicit val system = ActorSystem("Appka")
```
Comment (raboof) [inline bootstrap-demo/kubernetes-api/src/main/scala/akka/cluster/bootstrap/DemoApp.scala]: Appka?

File: bootstrap-demo/kubernetes-dns/src/main/scala/akka/cluster/bootstrap/KubernetesHealthChecks.scala [hunk 5]
```diff
@@ -17,8 +17,10 @@ class KubernetesHealthChecks(system: ActorSystem) {
 
   //#health
   private val readyStates: Set[MemberStatus] = Set(MemberStatus.Up, MemberStatus.WeaklyUp)
+
+  // Removed is the initial state before joining. See https://github.com/akka/akka/issues/25663
   private val aliveStates: Set[MemberStatus] =
-    Set(MemberStatus.Joining, MemberStatus.WeaklyUp, MemberStatus.Up, MemberStatus.Leaving, MemberStatus.Exiting)
+    Set(MemberStatus.Joining, MemberStatus.WeaklyUp, MemberStatus.Up, MemberStatus.Leaving, MemberStatus.Exiting, MemberStatus.Removed)
```
Comment (raboof) [inline bootstrap-demo/kubernetes-dns/src/main/scala/akka/cluster/bootstrap/KubernetesHealthChecks.scala]: Ah you beat me to it (#315)

File: discovery-kubernetes-api/src/main/resources/reference.conf [hunk 6]
```diff
@@ -17,14 +17,25 @@ akka.discovery {
 
     # Namespace to query for pods
     pod-namespace = "default"
+
+    # Picks up namespace from $NAMESPACE env var if set. Add the following to the container spec to avoid having to
+    # hard code this
```
Comment (raboof) [inline discovery-kubernetes-api/src/main/resources/reference.conf]: :+1:

File: bootstrap-demo/kubernetes-api/kubernetes/akka-cluster.yml [hunk 7]
```diff
@@ -28,13 +28,18 @@ spec:
...
+        - name: management
           containerPort: 8558
           protocol: TCP
+#namespace
+        env:
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
```
Comment (raboof) [inline bootstrap-demo/kubernetes-api/kubernetes/akka-cluster.yml]: neat

**Pull request merged**